airbyte-commons-server/src/main/java/io/airbyte/commons/server/handlers/WebBackendConnectionsHandler.java

```diff
@@ -266,8 +263,25 @@ static private WebBackendConnectionListItem buildWebBackendConnectionListItem(
   @VisibleForTesting
   static SchemaChange getSchemaChange(
                                       final ConnectionRead connectionRead,
-                                      final Optional<UUID> currentSourceCatalogId,
-                                      final Optional<ActorCatalogFetchEvent> mostRecentFetchEvent) {
+                                      final CatalogDiff diff) {
+    if (connectionRead == null || diff == null) {
+      return SchemaChange.NO_CHANGE;
+    }
+
+    if (connectionRead.getBreakingChange() != null && connectionRead.getBreakingChange()) {
+      return SchemaChange.BREAKING;
+    }
+
+    if (!diff.getTransforms().isEmpty()) {
+      return SchemaChange.NON_BREAKING;
+    }
+
+    return SchemaChange.NO_CHANGE;
+  }
+
+  static SchemaChange getSchemaChangeForList(final ConnectionRead connectionRead,
```

Comment:
Should this be `@VisibleForTesting` as well or private?

Reply:
This method is still here to use the old method for the connection list page. I didn't want to calculate a catalog diff for every connection in a workspace since it's kind of an expensive operation, and it's less impactful to see an incorrect schema change notification on the list page vs the connection page itself.